master/internal/sproto/resources.go

```diff
@@ -155,7 +155,10 @@ func NewResourcesFailure(
 
 func (f ResourcesFailure) Error() string {
 	if f.ExitCode == nil {
-		return fmt.Sprintf("%s: %s", f.FailureType, f.ErrMsg)
+		if len(f.ErrMsg) > 0 {
```

Comment:
this is fine, but i think the bigger issue what why is the errmsg empty, and is there anyway to get a better message from dispatcher about what happened to the slurm job (that doesnt just parse the job log, or print it)?

Reply:
Here is the PR https://github.com/determined-ai/determined-ee/pull/583,  "Reduce verbosity of failure messages"

Reply:
Here is an example. Note the errmsg are displayed above (or in experiment logs for experiment run)

% det command run exit 77
[2023-03-09T15:47:44.819713Z]          || INFO: Scheduling Command (vastly-caring-gnat) (id: 2b03690a-159a-4e6a-86d2-11150d311441)
[2023-03-09T15:47:45.030003Z]          || INFO: Command (vastly-caring-gnat) was assigned to an agent
[2023-03-09T15:47:56.423225Z]          || INFO: HPC Job ID: 24108
[2023-03-09T15:48:16.836878Z]          || ERROR: INFO:    Using cached SIF image
WARNING: Overriding HOME environment variable with APPTAINERENV_HOME is not permitted
INFO: Setting workdir to /run/determined/workdir
INFO: executing /run/determined/command-entrypoint.sh exit 77
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
jupyterlab-server 2.19.0 requires jsonschema>=4.17.3, but you have jsonschema 3.2.0 which is incompatible.
srun: error: node002: task 0: Exited with exit code 77
srun: launch/slurm: _step_signal: Terminating StepId=24108.0

[2023-03-09T15:48:16.856735Z]          || ERROR: Command (vastly-caring-gnat) was terminated: allocation failed: resources failed with non-zero exit code
Task log stream ended. To reopen log stream, run: det task logs -f 2b03690a-159a-4e6a-86d2-11150d311441